### PR TITLE
Prefix all extension URLs with base_url (including static)

### LINKED ISSUE #282: Static content with base_url
Current behavior when you run with a base_url with `jupyter server --ServerApp.jpserver_extensions="{'simple_ext1': True}" --ServerApp.base_url=/my_prefix`

- Custom handler are served behind `/my_prefix` the http://localhost:8888/my_prefix/simple_ext1/default 
- Static content is not served behind `/my_prefix` http://localhost:8888/static/simple_ext1/home.html 

I can understand static content is nicely segregated by `/static/extension_name/...` but with reverse proxy, deployments typically serve all content (static + dynamic) behind a fixed prefix `/my_prefix` which is not the case here.

Do we keep this like that or do we evolve to `/my_prefix/static/extension_name/...`?
Based on the decision, we will need to fix JupyterLab.

cc/  @Zsailer @blink1073

### PR DESCRIPTION
**Pull request by Zsailer**
Fixes #282.

Adds the base_url as a prefix to all extensions URLs, including the static namespaced URLs. 

I've also added a test to check that the static URL is properly prefixed.

### CODE REVIEW

File: jupyter_server/extension/application.py [hunk 1]
```diff
@@ -288,6 +289,7 @@ def _prepare_handlers(self):
         # Add static endpoint for this extension, if static paths are given.
         if len(self.static_paths) > 0:
             # Append the extension's static directory to server handlers.
+            print(self.static_url_prefix)
```
Comment (blink1073) [inline jupyter_server/extension/application.py:292]: debug

File: jupyter_server/extension/serverextension.py [hunk 2]
```diff
@@ -295,6 +295,8 @@ def list_server_extensions(self):
             {"user": False, "sys_prefix": True},
             {"user": False, "sys_prefix": False}
         )
+        self.log.info
```
Comment (blink1073) [inline jupyter_server/extension/serverextension.py:298]: wip?

File: jupyter_server/extension/application.py (hunk 1 above)
Comment (Zsailer) [reply jupyter_server/extension/application.py:292]: 😆 case of the Mondays

File: jupyter_server/extension/serverextension.py (hunk 2 above)
Comment (Zsailer) [reply jupyter_server/extension/serverextension.py:298]: Oops, leftover from another PR.

**Review by echarles — APPROVED** [review]
Tested with the example and works as expected. Thx a lot @Zsailer for this fixe and for the unit test. Code LGTM. Merging.

File: jupyter_server/extension/serverextension.py [hunk 3]
```diff
@@ -45,7 +45,7 @@ def _get_config_dir(user=False, sys_prefix=False):
     return extdir
 
 
-def _get_extmanager_for_context(user=False, sys_prefix=False):
+def _get_extmanager_for_context(write_dir="jupyter_server_config.d", user=False, sys_prefix=False):
```
Comment (vidartf) [inline jupyter_server/extension/serverextension.py:48]: Sorry for commenting on an old, merged PR, but what is this argument for? It is unused and undocumented in docstring.

Comment (Zsailer) [reply jupyter_server/extension/serverextension.py:48]: Good catch, @vidartf. I've updated this function to use the argument and added a docstring in #297.

**Pull request merged**